pkg/operator/controller/dateutil.go

```diff
@@ -0,0 +1,32 @@
+package controller
+
+import (
+	"github.com/gorhill/cronexpr"
+	"time"
+)
+
+// nextCronDuration check if next cron activation time has exceeded if so return true
+// if activation time has not reached return false and remaining time
+// in case it failed to parse cron expression return error
+func nextCronDuration(cronString string, creationTime time.Time) (time.Duration, error) {
+	expr, err := cronexpr.Parse(cronString)
+	if err != nil {
+		return time.Duration(0), err
+	}
+	return timeToExpiration(expr.Next(creationTime)), nil
+}
+
+//durationExceeded  check if duration is now meaning zero
+func durationExceeded(duration time.Duration) bool {
+	return duration.Nanoseconds() <= 0
+}
+
+//timeToExpiration  return the duration between time to expiration
+func timeToExpiration(expiresAt time.Time) time.Duration {
+	return expiresAt.Sub(time.Now())
```

Comment:
We have abstract data type `ext.Clock` to "stop the time" and facilitate testing code that manipulates data & time. Instead of using time.Now() directly I'd consider ext.Clock